packages/orchestration/src/fixtures/zoe-tools.contract.js

```diff
@@ -0,0 +1,105 @@
+import { makeSharedStateRecord } from '@agoric/async-flow';
+import { InvitationShape } from '@agoric/zoe/src/typeGuards.js';
+import { E } from '@endo/far';
+import { M } from '@endo/patterns';
+import { withOrchestration } from '../utils/start-helper.js';
+import { prepareChainHubAdmin } from '../exos/chain-hub-admin.js';
+import * as flows from './zoe-tools.flows.js';
+import fetchedChainInfo from '../fetched-chain-info.js';
+
+const { values } = Object;
+
+/**
+ * @import {TimerService} from '@agoric/time';
+ * @import {LocalChain} from '@agoric/vats/src/localchain.js';
+ * @import {NameHub} from '@agoric/vats';
+ * @import {Remote} from '@agoric/vow';
+ * @import {Zone} from '@agoric/zone';
+ * @import {AssetInfo} from '@agoric/vats/src/vat-bank.js';
+ * @import {CosmosInterchainService} from '@agoric/orchestration';
+ * @import {OrchestrationTools} from '../utils/start-helper.js';
+ */
+
+/**
+ * @typedef {{
+ *   localchain: Remote<LocalChain>;
+ *   orchestrationService: Remote<CosmosInterchainService>;
+ *   storageNode: Remote<StorageNode>;
+ *   timerService: Remote<TimerService>;
+ *   agoricNames: Remote<NameHub>;
+ * }} OrchestrationPowers
+ */
+
+/**
+ * @param {ZCF} zcf
+ * @param {OrchestrationPowers & {
+ *   marshaller: Marshaller;
+ * }} privateArgs
+ * @param {Zone} zone
+ * @param {OrchestrationTools} tools
+ */
+const contract = async (
+  zcf,
+  privateArgs,
+  zone,
+  { chainHub, orchestrateAll, zoeTools },
+) => {
+  const contractState = makeSharedStateRecord(
+    /** @type {{ account: OrchestrationAccount<any> | undefined }} */ {
+      localAccount: undefined,
+    },
+  );
+
+  const creatorFacet = prepareChainHubAdmin(zone, chainHub);
+
+  const orchFns = orchestrateAll(flows, {
+    zcf,
+    contractState,
+    zoeTools,
+  });
+
+  const assets = /** @type {AssetInfo[]} */ (
+    await E(E(privateArgs.agoricNames).lookup('vbankAsset')).values()
+  );
+  for (const chainName of ['agoric', 'cosmoshub']) {
+    chainHub.registerChain(chainName, fetchedChainInfo[chainName]);
+  }
+  for (const brand of values(zcf.getTerms().brands)) {
+    const info = assets.find(a => a.brand === brand);
+    if (info) {
+      chainHub.registerAsset(info.denom, {
```

Comment:
chainHub has a reference to agoricNames. should it do this itself?

Reply:
That would preferable IMO.

My understanding is that:
 - `withOrchestration` provides a new ChainHub when invoked
 - calls like `getChainInfo`, `getConnectionInfo` reach out to `agoricNames` if data isn't cached, so the "emptiness" is not felt
 - `getAsset` and `getDenom` do not fall back to `agoricNames` if something isn't found, so a registration is necessary

I asked about plans [here](https://github.com/Agoric/agoric-sdk/issues/9752#issuecomment-2338685078), and the ticket still has a **needs-design** label. 

Happy to add a TODO comment above this block, [like here](https://github.com/Agoric/agoric-sdk/blob/e66135307081331502a413cab2d6d1563c982a16/packages/orchestration/test/examples/staking-combinations.test.ts#L87-L94), if we do not expect this to be the status quo.

Reply:
yeah, i think it still does need design. I didn't see your IBIS which is very good to move that along.

Meanwhile, an `// UNTIL https://github.com/Agoric/agoric-sdk/issues/9752` would make clear that this is a pattern that will change and also give us an easy thing to query for updating as we resolve 9752.